doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-kernel.rst

```diff
@@ -0,0 +1,69 @@
+.. _tests-integration-testing-teuthology-workflow:
+
+Integration Tests for Kernel Development
+========================================
+
+
+.. kernel-cephfs:
+
+CephFS
+------
+
+The ``fs`` suite runs various kernels as described by the `kernel yaml
+fragments`_. These are symbolically linked by other sub-suites under the ``fs``
+suite.
+
+The structure of the matrix of fragments allows for testing the following configurations:
+
+* The "stock" kernel on RHEL 8 (i.e. the kernel that ships with it).
+
+* The `testing branch`_ by the kernel development team which represents the
+  patches undergoing active testing. These patches may or may not be in the next
+  upstream kernel release and include a mix of CephFS or kRBD changes. For the
+  testing kernel, we test with whatever distributions are specified by the
+  sub-suite. For example, the ``fs:functional`` sub-suite uses a random selection
+  of the `supported random distros`_.
+
+
+
+
+Testing custom kernels
+----------------------
+
+If you have a kernel branch on `ceph-client.git`_ and built using shaman, then
```

Comment:
`have built` or `build` instead of `built` perhaps?

Reply:
done, thanks